sdk/include/platform/sunburst/platform-spi.hh

```diff
@@ -0,0 +1,175 @@
+#pragma once
+#include <cdefs.h>
+#include <stdint.h>
+
+/**
+ * A Simple Driver for the Sonata's SPI.
+ *
+ * Documentation source can be found at:
+ * https://github.com/lowRISC/sonata-system/blob/1a59633d2515d4fe186a07d53e49ff95c18d9bbf/doc/ip/spi.md
+ *
+ * Rendered documentation is served from:
+ * https://lowrisc.org/sonata-system/doc/ip/spi.html
+ */
+struct SonataSpi
+{
+	/// Interrupt State Register
+	uint32_t interruptState;
+	/// Interrupt Enable Register
+	uint32_t interruptEnable;
+	/// Interrupt Test Register
+	uint32_t interruptTest;
+	/// Configuration register. Controls how the SPI block transmits and
+	/// receives data. This register can only be modified whilst the SPI block
+	/// is idle.
+	uint32_t configuration;
+	/// Controls the operation of the SPI block. This register can
+	/// only be modified whilst the SPI block is idle.
+	uint32_t control;
+	/// Status information about the SPI block
+	uint32_t status;
+	/// When written begins an SPI operation.
```

Comment:
This took a few attempts for me to parse.  Maybe 'Writes to this begin an SPI operation'?